src/sentry/incidents/logic.py

```diff
@@ -91,9 +92,16 @@ def create_incident(
             alert_rule=alert_rule,
         )
         if projects:
-            IncidentProject.objects.bulk_create(
-                [IncidentProject(incident=incident, project=project) for project in projects]
-            )
+            incident_projects = [
+                IncidentProject(incident=incident, project=project) for project in projects
+            ]
+            IncidentProject.objects.bulk_create(incident_projects)
+            # `bulk_create` doesn't send `post_save` signals, so we manually fire them here.
+            for incident_project in incident_projects:
+                post_save.send(
+                    sender=type(incident_project), instance=incident_project, created=True
```

Comment:
Would using `incident_project.__class__` be another way around shadowing `type()`